vendor/darwin/Foundation/NSString.odin

```diff
@@ -1,39 +1,12 @@
 package objc_Foundation
 
+import CF "core:sys/darwin/CoreFoundation"
+
 foreign import "system:Foundation.framework"
 
 @(objc_class="NSString")
 String :: struct {using _: Copying(String)}
 
-StringEncoding :: enum UInteger {
-	ASCII             = 1,
-	NEXTSTEP          = 2,
-	JapaneseEUC       = 3,
-	UTF8              = 4,
-	ISOLatin1         = 5,
-	Symbol            = 6,
-	NonLossyASCII     = 7,
-	ShiftJIS          = 8,
-	ISOLatin2         = 9,
-	Unicode           = 10,
-	WindowsCP1251     = 11,
-	WindowsCP1252     = 12,
-	WindowsCP1253     = 13,
-	WindowsCP1254     = 14,
-	WindowsCP1250     = 15,
-	ISO2022JP         = 21,
-	MacOSRoman        = 30,
-
-	UTF16             = Unicode,
-
-	UTF16BigEndian    = 0x90000100,
-	UTF16LittleEndian = 0x94000100,
-
-	UTF32             = 0x8c000100,
-	UTF32BigEndian    = 0x98000100,
-	UTF32LittleEndian = 0x9c000100,
-}
-
```

Comment:
BTW, are these correct? I see the following in Xcode:
```swift
/* Identifier for character encoding; the values are the same as Text Encoding Converter TextEncoding.
*/
public typealias CFStringEncoding = UInt32

/* Platform-independent built-in encodings; always available on all platforms.
   Call CFStringGetSystemEncoding() to get the default system encoding.
*/
public var kCFStringEncodingInvalidId: UInt32 { get }
public enum CFStringBuiltInEncodings : CFStringEncoding, @unchecked Sendable {

    
    case macRoman = 0

    case windowsLatin1 = 1280 /* ANSI codepage 1252 */

    case isoLatin1 = 513 /* ISO 8859-1 */

    case nextStepLatin = 2817 /* NextStep encoding*/

    case ASCII = 1536 /* 0..127 (in creating CFString, values greater than 0x7F are treated as corresponding Unicode value) */

    case unicode = 256 /* kTextEncodingUnicodeDefault  + kTextEncodingDefaultFormat (aka kUnicode16BitFormat) */

    case UTF8 = 134217984 /* kTextEncodingUnicodeDefault + kUnicodeUTF8Format */

    case nonLossyASCII = 3071 /* 7bit Unicode variants used by Cocoa & Java */

    
    public static var UTF16: CFStringBuiltInEncodings { get } /* kTextEncodingUnicodeDefault + kUnicodeUTF16Format (alias of kCFStringEncodingUnicode) */

    case UTF16BE = 268435712 /* kTextEncodingUnicodeDefault + kUnicodeUTF16BEFormat */

    case UTF16LE = 335544576 /* kTextEncodingUnicodeDefault + kUnicodeUTF16LEFormat */

    
    case UTF32 = 201326848 /* kTextEncodingUnicodeDefault + kUnicodeUTF32Format */

    case UTF32BE = 402653440 /* kTextEncodingUnicodeDefault + kUnicodeUTF32BEFormat */

    case UTF32LE = 469762304 /* kTextEncodingUnicodeDefault + kUnicodeUTF32LEFormat */
}
```

Reply:
and these in CFString header: https://opensource.apple.com/source/CF/CF-1153.18/CFString.h.auto.html

Reply:
To answer my own question, it doesn't seem like the ones I linked to are correct. I'm getting `024-04-05 12:53:52.400 cocoa[34665:824994] Incorrect NSStringEncoding value 0x8000100 detected. Assuming NSASCIIStringEncoding. Will stop this compatibility mapping behavior in the near future.` when trying to create an NSString to use an NSWindow title with
```
window_title := NS.String.alloc()->initWithOdinString("Cocoa window 😛")
window->setTitle(window_title)
```
I still wonder why though and where there the `UTF8              = 4,` enum values comes from?

Reply:
I found this documentation that matches up with the current values: https://developer.apple.com/documentation/foundation/1497293-string_encodings/nsutf8stringencoding

And then these docs for the CoreFoundation variants: https://developer.apple.com/documentation/corefoundation/cfstringbuiltinencodings

Now there is a distinction between CoreFoundation (CFString) and Foundation (NSString).

The values you have here, are for CoreFoundation, while the values we have (where utf8 is 4) is for Foundation.

So 2 different enumerations are required here.

Reply:
Ah, yes, that makes sense, thanks. I moved `StringEncoding` back to `NSString.odin` and added `StringBuiltInEncodings` and `StringEncodings` to `CFString.odin`.